### DISCUSSION TITLE
Comparison Error while deploying with k3s

### DISCUSSION THREAD

**Discussion by prabormukherjee** [k3s-io/k3s#4112] [Q&A]
I am trying to deploy an argocd application, but got the following error
```ComparisonErrorrpc error: code = Unknown desc = Get "https://github.com/prabormukherjee/Techtrends/info/refs?service=git-upload-pack": dial tcp: lookup github.com on 10.43.0.10:53: server misbehaving```

The build was successful: `$ kubectl apply -f argocd/helm-techtrends-prod.yaml`

> application.argoproj.io/techtrends-prod created  

`kubectl get application -n argocd`

> NAME     SYNC STATUS     HEALTH STATUS
techtrends-prod    Unknown    Healthy  

But when I am trying to see the deployment from `localhost:30007`, it's showing that error(mentioned above).  
If you want to check any yaml file, you can find them in the repo, [here](https://github.com/prabormukherjee/Techtrends). For yaml of `argocd`, check the `argocd/` (there I have `helm-techtrends-prod.yaml` and `helm-techtrends-staging.yaml`, that I want to deploy) and yaml of `helm`, check in `helm/`.   
I run these commands for debugging from multiple suggestions. [Here](https://ufile.io/sc03w92u) is the file, you can download.  
Upon investigating, it looks like some DNS resolver issue but I have no problem regarding proxy, firewall or something like this. Also I'm using `coredns` (not `kubedns`), which I think I the default for k3s. It has `Labels: k8s-app=kube-dns`  
If you know the issue or fix, please help me.

**Comment by prabormukherjee** [comment]
Here is the error message if it helps: 
`kubectl logs --namespace=kube-system -l k8s-app=kube-dns`
> [ERROR] plugin/errors: 2 github.com. A: read udp 10.42.0.31:41554->8.8.8.8:53: i/o timeout
[ERROR] plugin/errors: 2 github.com. AAAA: read udp 10.42.0.31:38547->8.8.8.8:53: i/o timeout
[ERROR] plugin/errors: 2 github.com. AAAA: read udp 10.42.0.31:57930->8.8.8.8:53: read: no route to host
[ERROR] plugin/errors: 2 github.com. A: read udp 10.42.0.31:53405->8.8.4.4:53: i/o timeout
[ERROR] plugin/errors: 2 github.com. AAAA: read udp 10.42.0.31:36083->8.8.8.8:53: i/o timeout
[ERROR] plugin/errors: 2 github.com. A: read udp 10.42.0.31:60495->8.8.8.8:53: i/o timeout
[ERROR] plugin/errors: 2 github.com. AAAA: read udp 10.42.0.31:52176->8.8.8.8:53: read: no route to host
[ERROR] plugin/errors: 2 github.com. A: read udp 10.42.0.31:52970->8.8.4.4:53: i/o timeout
[ERROR] plugin/errors: 2 github.com. A: read udp 10.42.0.31:54987->8.8.8.8:53: read: no route to host
[ERROR] plugin/errors: 2 github.com. AAAA: read udp 10.42.0.31:43776->8.8.8.8:53: i/o timeout

**Comment by brandond** [comment]
K3s will generate a replacement resolv.conf that points at 8.8.8.8 for upstream dns if it finds that your system /etc/resolv.conf contains loopback, broadcast, or ipv6 upstreams. If your node cannot reach 8.8.8.8 (google DNS) then obviously that won't work.

Your best bet is to fix your system /etc/resolv.conf, or create one specifically for use by k3s and specify it with the `--resolv-conf=/path/to/resolv.conf` flag.

**Reply by prabormukherjee** [reply]
Thanks for your reply. I checked the file. The nameserver is pointing to 127.0.0.1.. I also tried manually changing it to 8.8.8.8 but that also didn't help. I can ping google.com, so doesn't looks like firewall or proxy issue. What i should check again?
UPDATE: It was `127.0.0.53`

**Reply by brandond** [reply]
127.x usually means you're using systemd-resolved. What do you have in `/run/systemd/resolve/resolv.conf` ?

**Reply by prabormukherjee** [reply]
It has a nameserver 192.168.43.1
